lib/appcanary/assert.rb

```diff
@@ -0,0 +1,51 @@
+require "json"
+
+module Appcanary
+  class << self
+    def frequencies(arr)
+      arr.inject({}) do |freqs, i|
+        freqs[i] ||= 0
+        freqs[i] += 1
+        freqs
+      end
+    end
+
+    def criticalities(response)
+      frequencies response["included"]
+        .map { |vuln| vuln["attributes"]["criticality"] }
+    end
+
+    def vulnerable?
+      ship_gemfile :check do |response|
+        !! response["meta"]["vulnerable"]
```

Comment:
Minor style note:

I personally dislike "!!", because I prefer being maximally explicit instead of typecasting truthy values indirectly.

What are we guarding against and could we check that instead?

```
(response["meta"]["vulnerable"] == "true" || response["meta"]["vulnerable"] == true)
```

has a redundant feel to it but we make our assumptions explicit?